### PR TITLE
Support more scenarios with 'replace () with {}' code action

### PR DESCRIPTION
**Pull request by kpodsiad**
Aims to close https://github.com/scalameta/metals-feature-requests/issues/233.

I've found a scenario when rewrite parens action didn't work. Then I found an issue created by @agilesteel so I fed two birds with one seed :D

### CODE REVIEW

**Comment by kpodsiad** [comment]
Now there is code action for
```scala
  List(1,2).map { a =>
    <<>>println(a)
    a
  }
```
which effects in `println{a}` which is correct. There is no code action to rewrite braces for `map`

Although I think it really through, I'll be very grateful for more negative scenarios or cases when it's not working correctly. Especially braces to parens conversion

File: metals/src/main/scala/scala/meta/internal/metals/codeactions/RewriteBracesParensCodeAction.scala [hunk 1]
```diff
@@ -27,8 +41,13 @@ class RewriteBracesParensCodeAction(
     val path = params.getTextDocument().getUri().toAbsolutePath
     val range = params.getRange()
     val applyWithSingleFunction: Term.Apply => Boolean = {
-      case Term.Apply(_, List(_: Term.Function)) => true
-      case Term.Apply(_, List(Term.Block(List(_: Term.Function)))) => true
+      // exclude case when body has more than one line (is a Block) because it cannot be rewritten to parens
+      case Term.Apply(_, List(Term.Block(List(func: Term.Function))))
+          if func.body.isInstanceOf[Term.Block] =>
```
Comment (tgodzik) [inline metals/src/main/scala/scala/meta/internal/metals/codeactions/RewriteBracesParensCodeAction.scala:46]:
```suggestion
      case Term.Apply(_, List(Term.Block(List(Term.Function(_, _: Term.Block))))) =>
```

**Review by tgodzik — CHANGES_REQUESTED** [review]
Thanks a lot for the fixes! The tests seem to be failing unfortunately.

File: tests/unit/src/test/scala/tests/codeactions/ImportMissingSymbolLspSuite.scala [hunk 2]
```diff
@@ -17,6 +18,7 @@ class ImportMissingSymbolLspSuite
     s"""|${ImportMissingSymbol.title("Future", "scala.concurrent")}
         |${ImportMissingSymbol.title("Future", "java.util.concurrent")}
         |${CreateNewSymbol.title("Future")}
+        |${RewriteBracesParensCodeAction.toBraces}
```
Comment (tgodzik) [inline tests/unit/src/test/scala/tests/codeactions/ImportMissingSymbolLspSuite.scala:21]: Why does the action pop up here? It shouldn't. Also in the other places in this test suite.

**Review by tgodzik — CHANGES_REQUESTED** [review]

File: tests/unit/src/test/scala/tests/codeactions/ImportMissingSymbolLspSuite.scala (hunk 2 above)
Comment (kpodsiad) [reply tests/unit/src/test/scala/tests/codeactions/ImportMissingSymbolLspSuite.scala:21]: I think it should. Look what is selected: part of the `Term.Apply`, to be specific you selected following
```json
                    "type": "Term.Apply",
                    "fun": {
                      "type": "Term.Select", 
                      "qual": {
                        "type": "Term.Name",
                        "value": "Future" <<< here
                      },
                      "name": {
                        "type": "Term.Name",
                        "value": "successful"
                      }
```
I think here it's a valid usage, tbh I'm more concerned about more ambiguous cases. Maybe, because of that, we should contribute this code action only when the selection range is equal to zero (just cursor, no selection at all)?

File: metals/src/main/scala/scala/meta/internal/metals/codeactions/RewriteBracesParensCodeAction.scala [hunk 3]
```diff
@@ -15,56 +15,71 @@ import scala.meta.tokens.Token
...
-      case _ => false
-    }
-    trees
-      .findLastEnclosingAt[Term.Apply](
-        path,
-        range.getStart(),
-        applyWithSingleFunction
-      )
+    val applyTree =
+      if (range.getStart == range.getEnd)
```
Comment (kpodsiad) [inline metals/src/main/scala/scala/meta/internal/metals/codeactions/RewriteBracesParensCodeAction.scala:44]: @tgodzik what do you think about this condition? IMO it reduces the ambiguity of scenarios where the user selects multiple lines and sees "rewrite to braces" code action

File: metals/src/main/scala/scala/meta/internal/metals/codeactions/RewriteBracesParensCodeAction.scala [hunk 4]
```diff
@@ -92,4 +107,15 @@ class RewriteBracesParensCodeAction(
 object RewriteBracesParensCodeAction {
   val toParens = "Rewrite to parenthesis"
   val toBraces = "Rewrite to braces"
+  private val applyWithSingleFunction: Term.Apply => Boolean = {
```
Comment (tgodzik) [inline metals/src/main/scala/scala/meta/internal/metals/codeactions/RewriteBracesParensCodeAction.scala:110]: Can it be just a private method in the class? Any particular reason for moving it to the object?

File: tests/unit/src/test/scala/tests/codeactions/ConvertPatternMatchLspSuite.scala [hunk 5]
```diff
@@ -12,7 +13,7 @@ class PatternMatchRefactorLspSuite
        |  List(1,2).map { 
        |    _ match {
        |      case 1 => {
-       |        x = <<1>>
+       |        x = <<>>1
```
Comment (tgodzik) [inline tests/unit/src/test/scala/tests/codeactions/ConvertPatternMatchLspSuite.scala:16]: Is this change needed?

**Review by tgodzik — CHANGES_REQUESTED** [review]
Just some minor comments left, good job!

File: metals/src/main/scala/scala/meta/internal/metals/codeactions/RewriteBracesParensCodeAction.scala (hunk 3 above)
Comment (kpodsiad) [reply metals/src/main/scala/scala/meta/internal/metals/codeactions/RewriteBracesParensCodeAction.scala:43]: Sound great. Rewrite braces, convert pattern match, is that all which could benefit from that distinction?

Comment (tgodzik) [reply metals/src/main/scala/scala/meta/internal/metals/codeactions/RewriteBracesParensCodeAction.scala:43]: string interpolation actions also, really anything that has a single title

**Review by tgodzik — APPROVED** [review]
LGTM!
